src/ORM/DataObjectSchema.php

```diff
@@ -654,7 +655,12 @@ protected function buildSortDatabaseIndexes($class)
                     if ($table && strtolower($table ?? '') !== strtolower(DataObjectSchema::tableName($class) ?? '')) {
                         continue;
                     }
-                    if ($this->databaseField($class, $column, false)) {
+                    $fieldSpec = $this->databaseField($class, $column, false);
+                    if ($fieldSpec) {
+                        $dbField = Injector::inst()->create($fieldSpec, $column);
+                        if ($dbField instanceof DBText) {
+                            continue;
+                        }
```

Comment:
Do we need to instantiate the field? Could we use `is_a` check on value from `classForField`?

Reply:
The value from `classForField` is an injector specification and not a class name, so we do have to instantiate it first unfortunately.

Reply:
Ahh, nope, `DataObjectSchema::classForField()` gives you the `DataObject` that declares a field, not the `DBField` class that represents the field.

e.g. `$schema->classForField(MyPage::class, 'URLSegment')` would return `SiteTree::class`